File: EESoft/DataCenterVB/Backup/DataCenter-Network Service/MModule.vb
Imports System.Net
Imports System.Net.Sockets
Imports Microsoft.Win32
Module MModule
    'Forms
    Public FrmOptions As New OptionsFrm
    Public FrmMain As New MainFrm
    'Options
    Public DeletedFilesPath As String = ""
    Public GeneralPort As Int16
    'Public AutoStart As String
    Public FileIfo As String
    Public EnableCommand As Boolean = False
    Public Enum Commands 'this is the commands between the server and the client
        Ping ' command 0
        SendFile 'command 1
        ReciveFile 'command 2
        CreateFile 'command 3
        CreateFolder 'command 4
        RemoveFolder 'Command 5
        RemoveFile 'Command 6
        MoveFolder 'Command 7
    End Enum
    Public Enum RegKeyValues
        GeneralPort
        DeletedFilesPath
    End Enum
    Public Sub PrepareReg()
        Dim RegKey As RegistryKey = Registry.LocalMachine.OpenSubKey("SOFTWARE", True)
        Dim arr() As String
        Dim ForInt As Integer = 0
        arr = Registry.LocalMachine.OpenSubKey("SOFTWARE").GetSubKeyNames
        For ForInt = 0 To arr.Length - 1
            If arr(ForInt) = "Mena Data Center" Then
            End If
        Next
        RegKey.CreateSubKey("Mena Data Center")
    End Sub
    Public Sub SetRegValue(ByVal KeyValue As String, ByVal ValueName As RegKeyValues)
        PrepareReg()
        Registry.LocalMachine.OpenSubKey("SOFTWARE\Mena Data Center", True).SetValue(ValueName.ToString, KeyValue)
    End Sub
    Public Function GetRegValue(ByRef KeyValue As String, ByVal ValueName As RegKeyValues) As Boolean
        PrepareReg()
        Dim arr() As String = Registry.LocalMachine.OpenSubKey("SOFTWARE\Mena Data Center", False).GetValueNames
        Dim ForInt As Integer = 0
        For ForInt = 0 To arr.Length - 1
            If arr(ForInt) = ValueName.ToString Then
                KeyValue = Registry.LocalMachine.OpenSubKey("SOFTWARE\Mena Data Center", False).GetValue(ValueName.ToString)
                Return True
            End If
        Next
        Return False
    End Function
End Module

File: EESoft/DataCenter/DataCenter-Network Service/MModule.cs
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Collections;
using Microsoft.VisualBasic;
using System.Collections.Generic;
using System;
using System.Net;
using System.Net.Sockets;
using Microsoft.Win32;

namespace DataCenterNetworkService
{
	sealed class MModule
	{
		//Forms
		public static OptionsFrm FrmOptions = new OptionsFrm();
		public static MainFrm FrmMain = new MainFrm();
		//Options
		public static string DeletedFilesPath = "";
		public static short GeneralPort;
		//Public AutoStart As String
		public static string FileIfo = string.Empty;
        public static bool EnableCommand;
		public enum Commands //this is the commands between the server and the client
		{
			Ping, // command 0
			SendFile, //command 1
			ReciveFile, //command 2
			CreateFile, //command 3
			CreateFolder, //command 4
			RemoveFolder, //Command 5
			RemoveFile, //Command 6
			MoveFolder //Command 7
		}
		public enum RegKeyValues
		{
            GeneralPort,
            DeletedFilesPath
		}

        public static void PrepareReg()
        {
            RegistryKey RegKey = Registry.LocalMachine.OpenSubKey("SOFTWARE", true);
            string[] arr;
            arr = Registry.LocalMachine.OpenSubKey("SOFTWARE").GetSubKeyNames();
            for (int i = 0; i < arr.Length; i++)
            {
                if (arr[i] == "DataCenter")
                {
                    return;
                }
            }
            RegKey.CreateSubKey("DataCenter");
        }
        public static string GetRegValue(string ValueName)
        {
            PrepareReg();
            string ReturnMe = string.Empty;
            string[] arr = Registry.LocalMachine.OpenSubKey("SOFTWARE\\DataCenter", false).GetValueNames();
            for (int i = 0; i < arr.Length; i++)
            {
                if (arr[i] == ValueName)
                {
                    ReturnMe = (string)Registry.LocalMachine.OpenSubKey("SOFTWARE\\DataCenter", false).GetValue(ValueName);
                    break;
                }
            }
            return ReturnMe;
        }
		public static bool GetRegValue(ref string KeyValue, RegKeyValues ValueName)
		{
            
            foreach (string str in Registry.LocalMachine.OpenSubKey("SOFTWARE").GetSubKeyNames())
            {
                if (str == "DataCenter")
                {
                    string[] arr = Registry.LocalMachine.OpenSubKey("SOFTWARE\\DataCenter", false).GetValueNames();
                    int ForInt = 0;
                    for (ForInt = 0; ForInt <= arr.Length - 1; ForInt++)
                    {
                        if (arr[ForInt] == ValueName.ToString())
                        {
                            KeyValue = (string)(Registry.LocalMachine.OpenSubKey("SOFTWARE\\DataCenter", false).GetValue((string)(ValueName.ToString())));
                            return true;
                        }
                    }
                }
            }
            return false;
		}
	}
}
